API/Services/Tasks/VersionUpdaterService.cs

```diff
@@ -90,12 +90,19 @@ public async Task<IEnumerable<UpdateNotificationDto>> GetAllReleases()
         private UpdateNotificationDto CreateDto(GithubReleaseMetadata update)
         {
             if (update == null || string.IsNullOrEmpty(update.Tag_Name)) return null;
-            var version = update.Tag_Name.Replace("v", string.Empty);
-            var updateVersion = new Version(version);
+            var buildVersion = BuildInfo.Version;
+            var updateVersion = new Version(update.Tag_Name.Replace("v", string.Empty));
+            var currentVersion = buildVersion.ToString();
+
+            if (updateVersion.Revision == -1)
+            {
+                currentVersion = buildVersion.ToString()
```

Comment:
You're close, so it's like this:
```
var currentVersion = buildVersion.ToString();
currentVersion = currentVersion .SubString(0, buildVersion.LastIndexOf("."))
```
You don't want to keep calling toString because it keeps creating a temp string.

Reply:
Ah, okay. Sorry, still brushing up on my C#. Python has spoiled me, lol.